lib/auth/auth.go

```diff
@@ -6707,62 +6709,80 @@ func (a *Server) addAdditionalTrustedKeysAtomic(ctx context.Context, ca types.Ce
 // Keep this function in sync with lib/service/suite/suite.go:NewTestCAWithConfig().
 func newKeySet(ctx context.Context, keyStore *keystore.Manager, caID types.CertAuthID) (types.CAKeySet, error) {
 	var keySet types.CAKeySet
+
+	// Add SSH keys if necessary.
 	switch caID.Type {
-	case types.UserCA, types.HostCA:
-		sshKeyPair, err := keyStore.NewSSHKeyPair(ctx)
-		if err != nil {
-			return keySet, trace.Wrap(err)
-		}
-		tlsKeyPair, err := keyStore.NewTLSKeyPair(ctx, caID.DomainName)
+	case types.UserCA, types.HostCA, types.OpenSSHCA:
```

Comment:
You have to update `newKeySet` here to generate keys for your CA, otherwise your CA will just have no keys. If you're adding a CA here, hopefully you'll take a peek at `sshCAKeyPurpose` and realize you need to add your CA there, and then you'll realize you need to add a new `cryptosuites.KeyPurpose`. But yes, if you managed to add a new CA only to `newKeySet`, it would fail when calling `keyStore.NewSSHKeyPair(ctx, cryptosuites.KeyPurposeUnspecified)`, and you'd never be able to start an auth server, so we'd catch that pretty quick

If looking for places to update I'd recommend just grepping for another CA type, e.g. `types.SPIFFECA` only has 12 references and seems to have hit everything necessary